### PR TITLE
Add RemoveSuccess class, able to remove contact at specified index

### PR DESCRIPTION
**Pull request by lynnetteeee**
Currently unable to catch error when "yes"/"no" is not preceded by a "remove"

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/RemoveAbortion.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import seedu.address.model.Model;
+import seedu.address.logic.commands.exceptions.CommandException;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+public class RemoveAbortion extends RemoveConfirmation {
+    public static final String COMMAND_WORD = "no";
+    public static final String MESSAGE_REMOVE_PERSON_ABORTED = "Removal of contact aborted.";
```
Comment (kervyntan) [inline src/main/java/seedu/address/logic/commands/RemoveAbortion.java:11]: Nice use of constants

File: src/main/java/seedu/address/logic/commands/RemoveConfirmation.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    protected static final String MESSAGE_INVALID_DECISION_MAKING = "'yes' or 'no' is only used for decision making!"
+            + "Please start with a valid command word!";
+
+    // TO PROPERLY IMPLEMENT!!
+    // to check if previous command from command history is remove INDEX, as in other cases, typing in the command
+    // "yes" or "no" will not make sense
+    public boolean isValidConfirmation() {
+        return true;
+    }
+}
```
Comment (kervyntan) [inline src/main/java/seedu/address/logic/commands/RemoveConfirmation.java]: Appropriate abstraction of isValidConfirmation for future implementation

File: src/main/java/seedu/address/logic/parser/fuzzy/FuzzyCommandParser.java [hunk 3]
```diff
@@ -8,7 +8,7 @@
 public class FuzzyCommandParser {
     private static final BkTreeCommandMatcher<String> commandTree = new BkTreeCommandMatcher<>(Arrays.asList(
             "add", "edit", "delete", "remove", "clear", "find", "list", "exit", "help", "sort", "addbystep", "filter",
-            "duplicate", "overwrite"));
+            "duplicate", "overwrite", "yes", "no"));
```
Comment (kervyntan) [inline src/main/java/seedu/address/logic/parser/fuzzy/FuzzyCommandParser.java:11]: Nice meticulousness to add these constants to CommandTree!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -148,6 +148,12 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
         filteredPersons.setPredicate(predicate);
     }
 
+    // to update the filtered person list with a single person
+    public void updateSinglePersonList(Person person) {
+        requireNonNull(person);
+        filteredPersons.setPredicate(person::equals);
+    }
```
Comment (kervyntan) [inline src/main/java/seedu/address/model/ModelManager.java:155]: Good use of setPredicate with person::equals

File: src/main/java/seedu/address/logic/commands/RemoveSuccess.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+public class RemoveSuccess extends RemoveConfirmation {
+
+    public static final String COMMAND_WORD = "yes";
+
+    public static final String MESSAGE_REMOVE_PERSON_SUCCESS = "Successfully Removed Contact: %1$s";
+
+    private final int DEFAULT_INDEX = 0;
+
+    public RemoveSuccess() {
+    }
```
Comment (kervyntan) [inline src/main/java/seedu/address/logic/commands/RemoveSuccess.java:25]: Are there places using the empty constructor as of now?

**Review by kervyntan — COMMENTED** [review]
Looks ok overall, just issues with building gradle

File: src/main/java/seedu/address/logic/commands/RemoveConfirmation.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+public abstract class RemoveConfirmation extends Command {
+    protected static final String MESSAGE_INVALID_DECISION_MAKING = "'yes' or 'no' is only used for decision making!"
+            + "Please start with a valid command word!";
+
+    private final RemoveCommandParser parserHelper = new RemoveCommandParser();
+
+    // TO PROPERLY IMPLEMENT!! - done, left with slap
+    // to check if previous command from command history is remove INDEX, as in other cases, typing in the command
+    // "yes" or "no" will not make sense
+    public boolean isValidConfirmation() {
```
Comment (RunjiaChen) [inline src/main/java/seedu/address/logic/commands/RemoveConfirmation.java:20]: Perhaps this portion can be abstracted out a bit more? It is starting to look like an arrowhead

File: src/main/java/seedu/address/logic/commands/RemoveSuccess.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (!isValidConfirmation()) {
+            throw new CommandException(MESSAGE_INVALID_DECISION_MAKING);
+        }
+        Person personToRemove = model.getFilteredPersonList().get(DEFAULT_INDEX);
```
Comment (RunjiaChen) [inline src/main/java/seedu/address/logic/commands/RemoveSuccess.java:33]: Good practice of slap!

File: src/main/java/seedu/address/logic/commands/RemoveConfirmation.java (hunk 6 above)
Comment (lynnetteeee) [reply src/main/java/seedu/address/logic/commands/RemoveConfirmation.java:20]: Thanks for pointing out, will fix it!

File: src/main/java/seedu/address/logic/commands/RemoveSuccess.java (hunk 5 above)
Comment (lynnetteeee) [reply src/main/java/seedu/address/logic/commands/RemoveSuccess.java:25]: Nope, will remove!

**Review by RunjiaChen — APPROVED** [review]
LTGM, Good job!

**Pull request merged**